Makefile.BSD

```diff
@@ -34,8 +34,10 @@ ${D}/Makefile::
 	( cd ${D} && printf 'SRCS=' && ${MAKEBSD} -V SRCS | sed -e 's| cpusupport-config.h||' ) >> $@
 	( cd ${D} && printf 'IDIRS=' && ${MAKEBSD} -V IDIRS ) >> $@
 	( cd ${D} && printf 'LDADD_REQ=' && ${MAKEBSD} -V LDADD_REQ ) >> $@
+	( cd ${D} && printf 'SUBDIR_DEPTH=' && ${MAKEBSD} -V SUBDIR_DEPTH ) >> $@
 	if [ `cd ${D} && ${MAKEBSD} -V NOINST` ]; then		\
 		cat Makefile.prog |				\
+		    sed -e 's|<THISDIR>|${D}|' |		\
```

Comment:
Why not just make THISDIR a variable which is set like all the others?

Reply:
Also, I don't think what you have here even works since in the non-NOINST case you're not substituting?